docs/book/src/forc/dependencies.md

```diff
@@ -33,6 +33,21 @@ The following example adds a dependency with an `ipfs` source.
 custom_lib = { ipfs = "QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG" }
 ```
 
+For `registry` sources, `forc` will first resolve the source declared by its name and version. This is done using the forc.pub-index repo (located at `https://github.com/FuelLabs/forc.pub-index`). The package name and version is used to convert the declaration to an IPFS CID. The resolved IPFS CID is then used by either a local IPFS node or a public gateway to actually fetch the package.
```

Comment:
> The resolved IPFS CID is then used by either a local IPFS node or a public gateway to actually fetch the package.

It should be using only the forc-pub IPFS gateway, not local or public gateways.

Reply:
Hmm I agree that by default we should use forc-pub gateway instead of the public gateway we are using right now. 

But it should be still configurable as it is right now in my opinion. Because If we are forcing people to get it from forc-pub IPFS node only, than why we keep packages in IPFS in the first place? I think that makes it equivalent to keeping everything in a S3 only with extra complexity.

Reply:
I made it so that if the user is trying to rely on a public gateway we override it with fuel owned one. But if they are relying on a local node explicitly, we keep that as it is